tests/e2e_test.go

```diff
@@ -0,0 +1,610 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package tests
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanche-network-runner/rpcpb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/awm-relayer/config"
+	"github.com/ava-labs/awm-relayer/database"
+	"github.com/ava-labs/awm-relayer/messages/teleporter"
+	"github.com/ava-labs/awm-relayer/peers"
+	relayerEvm "github.com/ava-labs/awm-relayer/vms/evm"
+	"github.com/ava-labs/coreth/rpc"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/plugin/evm"
+	"github.com/ava-labs/subnet-evm/tests/utils"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	predicateutils "github.com/ava-labs/subnet-evm/utils/predicate"
+	warpPayload "github.com/ava-labs/subnet-evm/warp/payload"
+	"github.com/ava-labs/subnet-evm/x/warp"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
```

Comment:
How do people feel about using a dot-import for `gomega`? I find the assertions really hard to read with the package path in every statement at least twice.

Reply:
I'm fine dot importing `gomega` in this context. I'd want to avoid them almost anywhere else in the repo though, especially in the application code.

Reply:
Yes for sure :)

Reply:
thoughts on dot importing `ginkgo` as well? That seems to be the standard from ginkgo bootstrapping https://onsi.github.io/ginkgo/#bootstrapping-a-suite. No strong pref though

Reply:
I'd slightly lean against it. Doing it for `gomega` is great because it's very repetitive and hurts readability. In general I like being explicit about which package something is from, and I don't think `ginkgo` is getting in the way.

Reply:
makes sense, sgtm